superset/assets/src/dashboardList/DashboardList.tsx

```diff
@@ -0,0 +1,252 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { SupersetClient } from '@superset-ui/connection';
+import { t } from '@superset-ui/translation';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Button, Modal, Panel } from 'react-bootstrap';
+import ListView from 'src/components/ListView/ListView';
+import { FetchDataConfig } from 'src/components/ListView/types';
+import withToasts from 'src/messageToasts/enhancers/withToasts';
+
+import './DashboardList.less';
+
+const PAGE_SIZE = 25;
+
+interface Props {
+  addDangerToast: (msg: string) => void;
+}
+
+interface State {
+  dashboards: any[];
+  dashboard_count: number;
+  loading: boolean;
+  showDeleteModal: boolean;
+  deleteCandidate: any;
+}
+class DashboardTable extends React.PureComponent<Props, State> {
+  public static propTypes = {
+    addDangerToast: PropTypes.func.isRequired,
+  };
+
+  public state: State = {
+    dashboard_count: 0,
+    dashboards: [],
+    deleteCandidate: {},
+    loading: false,
+    showDeleteModal: false,
+  };
+
+  public columns = [
+    {
+      Cell: ({
+        row: {
+          original: { url, dashboard_title },
+        },
+      }: any) => <a href={url}>{dashboard_title}</a>,
+      Header: 'Dashboard',
+      accessor: 'dashboard_title',
+      filterable: true,
+      sortable: true,
+    },
+    {
+      Cell: ({
+        row: {
+          original: { changed_by_name, changed_by_url },
+        },
+      }: any) => <a href={changed_by_url}>{changed_by_name}</a>,
+      Header: 'Creator',
+      accessor: 'changed_by_fk',
+      sortable: true,
+    },
+    {
+      Cell: ({
+        row: {
+          original: { published },
+        },
+      }: any) => (
+          <span className='no-wrap'>{published ? 'True' : 'False'}</span>
+        ),
+      Header: 'Published',
+      accessor: 'published',
+      sortable: true,
+    },
+    {
+      Cell: ({
+        row: {
+          original: { changed_on },
+        },
+      }: any) => (
+          <span className='no-wrap'>{moment(changed_on).fromNow()}</span>
+        ),
+      Header: 'Modified',
+      accessor: 'changed_on',
+      sortable: true,
+    },
+    {
+      Cell: ({ row: { state, original } }: any) => {
+        const handleDelete = () => this.handleDashboardDeleteConfirm(original);
```

Comment:
On the MVC implementation if the user does not have permission to delete, the button is not rendered

The `_info` endpoint provides the user permissions on the class permission context